coral-common/src/main/java/com/linkedin/coral/common/utils/TypeDerivationUtil.java

```diff
@@ -59,6 +59,11 @@ public RelDataType getRelDataType(SqlNode sqlNode) {
       throw new RuntimeException("SqlValidator does not exist to derive the RelDataType for SqlNode: " + sqlNode);
     }
 
+    RelDataType fromNodeDataType = sqlValidator.getValidatedNodeTypeIfKnown(sqlNode);
```

Comment:
Why `fromNodeDataType`? How about `cachedDataType` or `dataType`?

Reply:
changed to `cachedDataType`